File: crates/topcoat-runtime/browser/src/expr/expr_lit.ts
import type { Interpreter } from "./interpreter";

export type ExprLit<T> = {
	type: "Lit";
	value: T;
};

export function interpret_expr_lit<T>(
	expr: ExprLit<T>,
	_interpreter: Interpreter,
): T {
	return expr.value;
}

File: crates/topcoat-interop/src/ast/expr/expr_lit.rs
use proc_macro2::TokenStream;
use quote::quote;

use super::Expr;

impl Expr {
    pub(super) fn expr_lit_to_tokens(lit: &syn::ExprLit) -> syn::Result<TokenStream> {
        let inner = &lit.lit;
        Ok(quote! { ::topcoat::interop::ExprValue::new(#inner) })
    }
}
